src/toffy/image_stitching.py

```diff
@@ -218,7 +258,87 @@ def stitch_images(
                 channels=[chan],
                 max_image_size=max_img_size,
             )
+            if img_size_scale:
+                image_data = rescale_stitched_array(image_data, img_size_scale)
+
             fname = os.path.join(stitched_subdir, chan + "_stitched.tiff")
             stitched = data_utils.stitch_images(image_data, num_cols)
-            current_img = stitched.loc["stitched_image", :, :, chan].values / scale
+            current_img = stitched.loc["stitched_image", :, :, chan] / intensity_scale
             image_utils.save_image(fname, current_img)
+
+
+def rescale_images(img_data, scale, save_path=None):
+    """Rescale image data to a desired shape
+    Args:
+        img_data (np.array): data to be reshaped, expected to be either 2 or 4 dimensions
+        scale (int): amount to scale the data up or down
+        save_path (str): the location to save the tiff file
+    Returns:
+        numpy.array: data reshaped according to scale value
+    """
+    # check data dimensions
+    if len(img_data.shape) != 2 and len(img_data.shape) != 4:
+        raise ValueError("Image data must have either 2 or 4 dimensions.")
+
+    if scale < 1 and img_data.shape[1] % scale != 0:
+        raise ValueError("Scale value less than 1 must be a factor of the image size.")
+
+    # preserve fov and channel dimensions if needed
+    if len(img_data.shape) == 4:
+        scale = (1, scale, scale, 1)
+
+    # rescale dimension data while preserving values
+    data_type = img_data.dtype
+    rescaled_data = transform.rescale(
+        img_data,
+        scale,
+        mode="constant",
+        preserve_range=True,
+        order=0,
+    )
+    rescaled_data = rescaled_data.astype(data_type)
+
+    # overwrite image tiff
+    if save_path:
+        io_utils.validate_paths(save_path)
+        image_utils.save_image(save_path, rescaled_data)
+
+    return rescaled_data
+
+
+def fix_image_resolutions(resolution_data, extraction_dir):
+    """Rescales any images that are a different resolution than the majority in the run
+    Args:
+        resolution_data (pd.DataFrame): details the fov names and resolutions
+        extraction_dir (str): path to the extracted images dir for the specific run
+    """
+    io_utils.validate_paths(extraction_dir)
+
+    # only check extracted fovs
+    extracted_fovs = io_utils.list_folders(extraction_dir, substrs="fov")
+    resolution_data = resolution_data[np.isin(resolution_data.fov, extracted_fovs)]
+
+    # identify majority value, subset data for other resolutions
+    resolutions = resolution_data["pixels / 400 microns"]
+    correct_res = Counter(resolutions).most_common(1)[0][0]
+    res_change = resolution_data[resolution_data["pixels / 400 microns"] != correct_res]
+
+    if len(res_change) == 0:
+        print("No resolution scaling needed for any FOVs in this run.")
```

Comment:
Might be a good idea just to `return` here. Although the subsequent loop wouldn't run anyways, it's more so for code clarity.